### PR TITLE
Add statistics python project to build pipeline

### PR DESCRIPTION
**Pull request by clairernovotny**

### CODE REVIEW

File: .pipelines/NuGetGallery-CI.yml [hunk 1]
```diff
@@ -126,6 +126,40 @@ extends:
...
+              PIPX_HOME: $(Agent.ToolsDirectory)\pipx
+            steps:
+              - checkout: self
+                fetchDepth: 1
+                clean: true
+                fetchTags: false
+                path: $(NuGetGalleryDirectory)
+              - task: UsePythonVersion@0
+                inputs:
+                  versionSpec: '3.10'
```
Comment (joelverhagen) [inline .pipelines/NuGetGallery-CI.yml:146]: it is possible to float this version? I don't know if we really care about fixing a version in this case. I'm just thinking of ways to reduce ongoing maintenance of this code area (python version and dependency versions).

File: .pipelines/NuGetGallery-CI.yml [hunk 2]
```diff
@@ -126,6 +126,40 @@ extends:
...
+              - script: |
+                  python -m pip install --user pipx
+                  pipx ensurepath
+                displayName: 'Install tools'
+              - script: |
+                  pipx install poetry
+                  poetry install
+                  poetry build
+                  poetry run pytest tests/ --cov loginterpretation --cov-report html
+                workingDirectory: $(NuGetGalleryPath)\python\StatsLogParser
```
Comment (joelverhagen) [inline .pipelines/NuGetGallery-CI.yml:156]: Is there a `PublishTestResults` step we should include? like we have for the .NET builds. This populates the Tests tab on the build which is nice.

Comment (joelverhagen) [reply .pipelines/NuGetGallery-CI.yml:156]: Oh I see the pytest results have already shown up in ADO. How does that work?

**Review by joelverhagen — APPROVED** [review]

File: .pipelines/NuGetGallery-CI.yml (hunk 2 above)
Comment (clairernovotny) [reply .pipelines/NuGetGallery-CI.yml:156]: No, that's done automatically by the pytest-azurepipelines plugin, and you'll notice that the results and code coverage are already there :)

File: .pipelines/NuGetGallery-CI.yml (hunk 1 above)
Comment (clairernovotny) [reply .pipelines/NuGetGallery-CI.yml:146]: Python is picky.... 3.10 is the version included on the Spark 3.4 pool. It already floats the minor version.

**Review by atomdrago — COMMENTED** [review]
Nuget

**Pull request merged**